tests/test_empiricalVariance.py

```diff
@@ -0,0 +1,149 @@
+        overscan2 = Box2I(Point2I(overscanSize[0] + 2*imageSize[0], 0), overscanSize)
+
+        leftBox = Box2I(overscan1.getMin(), Extent2I(overscan1.getWidth() + image1.getWidth(), height))
+        rightBox = Box2I(image2.getMin(), Extent2I(image2.getWidth() + overscan2.getWidth(), height))
+
+        target1 = Box2I(Point2I(0, 0), imageSize)
+        target2 = Box2I(Point2I(image1.getWidth(), 0), imageSize)
+
+        # Set the pixels
+        exposure = ExposureF(Box2I(Point2I(0, 0), Extent2I(imageSize[0]*2 + overscanSize[0]*2, height)))
+        yy = np.arange(0, height, 1, dtype=np.float32)
+        leftImage = ExposureF(exposure, leftBox)
+        leftImage.image.array[:] = baseValue + yy[:, np.newaxis]
+        rightImage = ExposureF(exposure, rightBox)
+        rightImage.image.array[:] = baseValue - yy[:, np.newaxis]
+
+        leftOverscan = ExposureF(exposure, overscan1)
+        leftOverscan.image.array += np.random.normal(0.0, self.sigma, leftOverscan.image.array.shape)
+        rightOverscan = ExposureF(exposure, overscan2)
+        rightOverscan.image.array += np.random.normal(0.0, self.sigma, leftOverscan.image.array.shape)
+        exposure.mask.array[:] = 0.0
+        exposure.variance.array[:] = np.nan
+
+        # Construct the detectors
+        amps = AmpInfoCatalog(AmpInfoTable.makeMinimalSchema())
+        makeAmplifier(amps, "left", target1, image1, overscan1, gain, readNoise, saturation)
+        makeAmplifier(amps, "right", target2, image2, overscan2, gain, readNoise, saturation)
+        ccdBox = Box2I(Point2I(0, 0), Extent2I(image1.getWidth() + image2.getWidth(), height))
+        ccd = Detector("detector", 1, SCIENCE, "det1", ccdBox, amps, Orientation(), Extent2D(1.0, 1.0), {})
+        exposure.setDetector(ccd)
+        header = PropertyList()
+        header.add("EXPTIME", 0.0)
+        exposure.getInfo().setVisitInfo(VisitInfo(header))
+
+        self.exposure = exposure
+        self.config = IsrTask.ConfigClass()
+
+        # Disable everything we don't care about
+        self.config.doBias = False
+        self.config.doDark = False
+        self.config.doFlat = False
+        self.config.doFringe = False
+        self.config.doDefect = False
+        self.config.doAddDistortionModel = False
+        self.config.doWrite = False
+        self.config.expectWcs = False
+        self.config.doLinearize = False
+        self.config.doCrosstalk = False
+        self.config.doBrighterFatter = False
+        self.config.doAttachTransmissionCurve = False
+
+        # Set the things that match our test setup
+        self.config.overscanFitType = "CHEB"
+        self.config.overscanOrder = 1
+        self.config.doEmpiricalReadNoise = True
+
+        self.task = IsrTask(config=self.config)
+
+    def tearDown(self):
+        del self.exposure
+
+    def testEmpiricalVariance(self):
+        results = self.task.run(self.exposure)
+        exposure = results.exposure
```

Comment:
Mildly confusing variable name (since there is a `self.exposure` floating around already, but this is an exposure created from the result of processing `self.exposure`). Perhaps call one raw and one post-ISR.